src/signer/src/main/java/ee/ria/xroad/signer/tokenmanager/token/SoftwareTokenWorker.java

```
@@ -197,6 +195,7 @@ protected void deleteCert(String certId) throws Exception {
         Signature signature = Signature.getInstance(SIGNATURE_ALGORITHM);
```

Seems to silently ignore the signature algorithm parameter and use the default.

---

This is previous code. Signature mechanism of the software token is not configured via _device.ini_ configuration file. Software token uses always hardcoded _NONEwithRSA_ algorithm.

---

Since you have introduced the signature algorithm as a parameter, maybe also the software token should fail if the argument is not supported (cf. HardwareTokenWorker).
 
It seems that the method expects a digest as the data (and at least BatchSigner seems to always calculate one), so it could check that the algorithm is one of SHA*withRSA (?).

---

Additional checks added.

---

There is something wrong with the checks, I'm now getting signer errors:
ee.ria.xroad.common.CodedException: InvalidDataToBeSigned: Invalid data to be signed (length: 83)

---

It seems that the algorithm is already checked in SignerUtil.createDataToSign, which also adds a prefix to the digest. Therefore the digest length is longer that expected by the checks. It also means that the additional check is redundant. Maybe there should be a comment explaining why the parameter can be ignored in this case.

---

Sorry, my bad. I removed paranoid and invalid data length check (digest length vs _DigestInfo_ ASN.1 structure length in case SHA\*withRSA algorithms). Let's check only signature algorithm there (in case SHA\*withRSAandMGF1 algorithms are unexpectedly used).